content/posts/0009-php-rearchitecure.md

```diff
@@ -0,0 +1,107 @@
+to add new functionality to the language family, by breaking down
+what used to be three larger buildpacks, into 10+ new buildpacks responsible
+for single parts of the build process. This may sound like a lot of new
+buildpacks, but small lighter-weight buildpacks are actually what we strive
+for in the project. This concept is well-described in the [Paketo Buildpacks
+Philosophy Part 1 blog
+post](https://blog.paketo.io/posts/buildpack-philosophy-part-1/).
+
+The GIF below gives an overview of how the buildpacks have been split up based
+off of their main job. The new buildpacks are split into three order groups,
+one for the Nginx use case, one for the HTTPD use case, and one for the PHP
+built-in server use case. The order groups in the buildpack can be found in the
+[PHP buildpack `buildpack.toml`
+file](https://github.com/paketo-buildpacks/php/blob/main/buildpack.toml).
+
+![Rewrite GIF](/images/posts/0009/rewrite.gif)
+
+
+#### API Upgrades
+The buildpacks have also been re-written to leverage the newest Paketo
+tooling, such as the [packit
+library](https://github.com/paketo-buildpacks/packit). Using the packit library
+allows the buildpacks to run on the latest versions of the upstream Cloud
+Native Buildpacks specifiction to make sure our buildpacks support the full
+range of new features in the buildpacks ecosystem. The new PHP buildpacks are
+built on [Buildpack API
+0.7](https://github.com/buildpacks/spec/blob/buildpack/v0.7/buildpack.md), as
+opposed to the API 0.4 that the old buildpacks used.
+
+#### SBOM support
+Because of the API upgrade, the buildpacks now include [Software Bill of
+Materials](https://paketo.io/docs/concepts/sbom/) (SBOM) support. The SBOM
+provides metadata about the dependencies associated with your application
+image, and can be used for vulnerability scanning. Check out the [Paketo
+documentation](https://paketo.io/docs/howto/sbom/) for how to access the SBOM.
+
+#### Maintaining the same use cases
+If you're not already acquainted with the previous version of the buildpack PHP
+buildpack, the following use cases are supported:
+
+*  Build PHP apps that are served with HTTPD
+*  Build PHP apps that are served with Nginx
+*  Build PHP apps that are served the PHP Built-in Server
+*  Leverage the PHP FastCGI Process Manager (FPM) to run alongside servers
+*  Build apps that use Composer as an application level package manager
+*  Configure and run PHP session handlers for Redis and Memcached instances
+
+The main difference in the user experience is that support for setting
+configuration through a `buildpack.yml` has been removed in favour of
+environment variable configuration. The configuration options are now detailed
+in the [Paketo PHP How-To documentation](https://paketo.io/docs/howto/php/),
+with more in-depth details of buildpack behaviour available in the [Paketo PHP
+Buildpack Reference
+documentation](https://paketo.io/docs/reference/php-reference/).
```

Comment:
This is a broader conversation than this PR, but I think we should likely improve how our env variables are documented. Keeping the docs as close to the code as possible (buildpack `README` for example) helps reduce drift between documentation and code. Aggregating the docs (paketo.io/docs) helps with discoverability of options, but the PHP reference page doesn't list all of the options. Plus, in the PHP language family we also have an RFC in which the env vars should be listed (https://github.com/paketo-buildpacks/rfcs/blob/main/text/php/0002-buildpack-yml-to-env-var.md).

Reply:
Agreed. I don't think it's a great idea to list everything here but it would be a MUCH better user experience to have everything in once place. I actually think every env variable configuration is listed in the How To documentaiton page. The Reference docs are more of an outline of how the internals of the buildpacks work.